src/topology/algebra/infinite_sum/basic.lean

```diff
@@ -313,6 +318,16 @@ lemma has_sum.compl_add {s : set β} (ha : has_sum (f ∘ coe : sᶜ → α) a)
   has_sum f (a + b) :=
 ha.add_is_compl is_compl_compl.symm hb
 
+lemma has_sum.sum_t {f : β ⊕ γ → α} {a_inl a_inr : α} (h_inl : has_sum (f ∘ sum.inl) a_inl)
+  (h_inr : has_sum (f ∘ sum.inr) a_inr) : has_sum f (a_inl + a_inr) :=
+has_sum.add_is_compl (set.is_compl_range_inl_range_inr)
+  (sum.inl_injective.has_sum_range_iff.2 h_inl) (sum.inr_injective.has_sum_range_iff.2 h_inr)
+
+lemma has_sum.option {f : option β → α} {a_some : α} (hf : has_sum (f ∘ option.some) a_some) :
```

Comment:
Please make it `protected`.

Reply:
Also, please add `summable` versions of your lemmas (with `iff` versions whenever we can apply `summable.comp_injective`).